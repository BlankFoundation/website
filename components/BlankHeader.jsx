import {
  NextLink,
  TWCleanHeader,
  TWFlex
} from '.'

const BlankHeader = ({ title, children }) => {
  return (
    <TWCleanHeader
      className='py-12 lg:px-24'
    >
      <NextLink href="/" passHref>
        <a>{title}</a>
      </NextLink>
      <TWFlex className="space-x-2">
        {children}
      </TWFlex>
    </TWCleanHeader>
  );
}

BlankHeader.defaultProps = {
  children: ['child1', 'child2', 'child3'].map(
    (child, index) => <div key={'child-' + index}>{child}</div>
  ),
  title: "Blank.Foundation"
}

export default BlankHeader;