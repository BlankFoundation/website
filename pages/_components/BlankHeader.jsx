import {
  TWCleanHeader,
  NextLink,
  TWFlex
} from '.'

const BlankHeader = ({ title, children }) => {
  return (
    <TWCleanHeader
      className='py-12 px-24'
    >
      <NextLink href="/" passHref>
        <a>{title}</a>
      </NextLink>
      <TWFlex className="space-x-2">
        {children}
      </TWFlex>
    </TWCleanHeader>
  );
}

BlankHeader.defaultProps = {
  children: ['child1', 'child2', 'child3'].map(
    (child, index) => <div key={'child-' + index}>{child}</div>
  ),
  title: "Blank.Foundation"
}

export default BlankHeader;