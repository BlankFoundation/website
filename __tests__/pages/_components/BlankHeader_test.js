import React from 'react';
import TestRenderer from 'react-test-renderer';
import BlankHeader from '../../../pages/_components/BlankHeader';

describe("BlankHeader", () => {
  it('renders correctly', () => {
    const rendered = TestRenderer.create(<BlankHeader />).toJSON();
    expect(rendered).toMatchSnapshot()
  });
});