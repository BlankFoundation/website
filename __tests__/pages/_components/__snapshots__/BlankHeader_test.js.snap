// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlankHeader renders correctly 1`] = `
<div
  className="py-12 p-6 bg-white text-black leading-10"
>
  <div
    className="px-24 flex flex-row justify-between"
  >
    Blank.Foundation
    <div
      className="space-x-2 flex flex-row justify-between"
    >
      <div>
        child1
      </div>
      <div>
        child2
      </div>
      <div>
        child3
      </div>
    </div>
  </div>
</div>
`;
